TESTING.md

```diff
@@ -0,0 +1,140 @@
+# Geeqie testing and validation patterns
+
+Geeqie incorporates a number of forms of validation, including functional tests,
+unit tests, and static analysis.  These tests are defined towards the end of the
+root `meson.build` file (search the file for `test(`).
+
+You can run all enabled suites using:
+
+```text
+meson test -C build
+```
+
+Three test suites exist: `functional`, `unit`, and `analysis`.  You can pick out
+particular suites to execute with commands like:
+
+```text
+meson test -C build --suite functional
+meson test -C build --suite analysis --suite unit
+```
+
+See the Unit tests section for how to enable unit tests.
+
+## Unit tests
+
+Unit tests live under `src/tests`.  Because they include a lot of extra code in
+the Geeqie binary, they must be manually enabled with a command like:
+
+```text
+meson setup -D unit_tests=enabled build
+```
+
+After that point, they can be executed with:
+
+```text
+meson test -C build -v --suite unit
+```
+
+Or you can run them by hand by starting geeqie with the `--run-unit-tests`
+argument:
+
```

Comment:
If this is a duplication of the meson option, maybe it would be better to exclude it.
Including the option in bash completions and in the --help output (but only when available) is probably possible but not worth the effort.
Certainly a single-character option is necessary.

Reply:
So, one thing I had intended that might not be clear is that I wrote `TESTING.md` for developers.  It is not intended for end users to see or pay attention to.

No, it only exists when the meson option is specified, but it is distinct.  It is intended only for triggering unit tests, and I would expect most developers to just let `meson test` handle things.  Here are the four cases:

1. `-Dunit_tests=disabled`, `--run-unit-tests` *not* specified: This is the common case.  Geeqie runs as normal.
2. `-Dunit_tests=disabled`, `--run-unit-tests` *specified*: Geeqie exits immediately with "Unit tests are not enabled in this build."
3. `-Dunit_tests=enabled`, `--run-unit-tests` *not* specified: Geeqie runs as normal, but unit test code is present in the binary.
4. `-Dunit_tests=enabled`, `--run-unit-tests` *specified*: Geeqie runs unit tests and exits immediately.

Because that option will always prevent Geeqie (as an image viewer) from starting, and is not intended for end users, I don't think it makes sense to include in `--help`.  Likewise, I don't think making it easier to specify (like with a single-character option) would be helpful.